src/OpalCompiler-Core/TemporaryVariable.class.st

```diff
@@ -20,7 +20,15 @@ TemporaryVariable >> copiedVarClass [
 
 { #category : #queries }
 TemporaryVariable >> definingNode [
-	^ scope node temporaries detect: [ :each | each name = name ]
+	^ scope node temporaries 
+		detect: [ :each | each name = name ]
+		ifNone: [ 
+			" ugly workaround to support temps defined by primitives"
+			| pragma |
+			pragma := scope node methodNode pragmas detect: [ :each | each isPrimitiveError ].
+			pragma ifNil: [ ^nil ].
+			^ RBVariableNode named: (pragma argumentAt: #error:) value asString
+			]
```

Comment:
This "error pragma" case really deserves a test and a comment with method example.
I do not know what it is about really.

Reply:
yes, I will do that... it needs some much better implementation, this is just a workaround to not have the rules fail on all those methods